profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java

```diff
@@ -40,15 +40,18 @@ public String serialize(RecordedStackTrace stack) {
         .toString();
   }
 
-  private StringBuilder serializeFrame(StringBuilder sb, RecordedFrame frame) {
-    RecordedMethod method = frame.getMethod();
+  private StringBuilder serializeFrame(StringBuilder sb, IMCFrame frame) {
+    IMCMethod method = frame.getMethod();
     if (method == null) {
       return maybeNewline(sb).append("\tat unknown.unknown(unknown)");
     }
     maybeNewline(sb);
-    String className = method.getType().getName();
-    String methodName = method.getName();
-    int lineNumber = frame.getInt("lineNumber");
+    String className = method.getType().getFullName();
+    String methodName = method.getMethodName();
+    Integer lineNumber = frame.getFrameLineNumber();
+    if (lineNumber == null) {
+      lineNumber = -1;
```

Comment:
[GDI spec ](https://github.com/signalfx/gdi-specification/blob/dd07dbd736dad4fd1b1a677473284799df2ee50c/specification/semantic_conventions.md#call-stack-format-for-text-data-format)says 0 when unknown.

Reply:
Thanks, will change. Note that previous parser returned `-1` when line number was not available

Reply:
Replaced all occurrences `-1`, please review.